Content.Server/Info/RulesManager.cs

```diff
@@ -1,15 +1,55 @@
-﻿using Content.Shared.Info;
-using Robust.Shared.IoC;
+﻿using System.Net;
+using Content.Server.Database;
+using Content.Shared.Info;
 using Robust.Shared.Network;
 
 namespace Content.Server.Info;
 
 public sealed class RulesManager : SharedRulesManager
 {
+    [Dependency] private readonly IServerDbManager _dbManager = default!;
     [Dependency] private readonly INetManager _netManager = default!;
 
+    private static DateTime LastValidReadTime => DateTime.UtcNow - TimeSpan.FromDays(60);
+
+    private readonly Dictionary<NetUserId, DateTime> _lastReadRulesCache = new();
+
     public void Initialize()
     {
+        _netManager.RegisterNetMessage<ShouldShowRulesPopupMessage>();
         _netManager.RegisterNetMessage<ShowRulesPopupMessage>();
+        _netManager.RegisterNetMessage<RulesAcceptedMessage>(OnRulesAccepted);
+        _netManager.Connected += OnConnected;
+    }
+
+    private async void OnConnected(object? sender, NetChannelArgs e)
+    {
+        if (IPAddress.IsLoopback(e.Channel.RemoteEndPoint.Address))
+        {
+            return;
+        }
+
+        var lastRead = await _dbManager.GetLastReadRules(e.Channel.UserId);
+        if (lastRead > LastValidReadTime)
+        {
+            return;
+        }
+
+        var message = _netManager.CreateNetMessage<ShouldShowRulesPopupMessage>();
+        _netManager.ServerSendMessage(message, e.Channel);
+    }
+
+    private void OnRulesAccepted(RulesAcceptedMessage message)
+    {
+        var date = DateTime.UtcNow;
+
+        if (_lastReadRulesCache.TryGetValue(message.MsgChannel.UserId, out var lastRead) &&
```

Comment:
This is the only place this cache is used. The only purpose it seems to serve is to avoid people spamming DB queries? I'm not sure this is something we need to bother with.

Reply:
Yes, that was the intent.
Removed it in commit [`097782d` (#7278)](https://github.com/space-wizards/space-station-14/pull/7278/commits/097782d66081dd4327de32d5280201b37214d1eb)